Added ability to edit records inline on the records management page.

tony-gaia opened the pull request:
Added a pipe for converting python date format to angular date format.

Added api function for updating record by field.

serge-gaia commented on src/app/pages/view/view-records/view-records.component.ts:
```diff
@@ -1,37 +1,61 @@
+
+        this.apiService.getSpecies().subscribe(
+            (species: string[]) => this.speciesDropdownItems =
+                this.speciesDropdownItems.concat(species.map(s => ({'label': s, 'value': s}))),
```
Just curious, any reasons to use `concat` instead of `=` ?
```
this.speciesDropdownItems = species.map(s => ({'label': s, 'value': s}))
```

serge-gaia commented on src/app/pages/view/view-records/view-records.component.ts:
```diff
@@ -40,55 +64,73 @@ export class ViewRecordsComponent implements OnInit {
+
+        this.exportURL = this.apiService.getRecordExportURL() +
+            Object.keys(recordParams).reduce(function(left, right){
+                left.push(right+'='+encodeURIComponent(recordParams[right]));return left;
```
whoot whoot! The `reduce` function! :-)

tony-gaia committed:
Added ability to edit records inline on the records management page.

Added a pipe for converting python date format to angular date format.

Added api function for updating record by field.

serge-gaia reviewed (commented):
Mainly a spelling mistake and maybe a refined date regex

tony-gaia committed:
Fixed spelling of ambiguous and updated regex for ambiguous date pattern.
